src/main/java/com/cburch/logisim/std/io/Joystick.java

```diff
@@ -87,10 +91,8 @@ public void paint(InstancePainter painter) {
       final var x = loc.getX();
       final var y = loc.getY();
       final var g = painter.getGraphics();
-      g.setColor(Color.WHITE);
-      g.fillRect(x - 20, y, 10, 10);
+      g.fillOval(x - 19, y + 1, 8, 8);
```

Comment:
what is the reason for not calling setColor()?

Reply:
I could call it with `Colour.BLACK` but this is pointless since black is the default.

Reply:
My concern is this: as some point we should make drawing code aware of user theme. In some cases it makes a difference (icons) in other it may not. So I am afraid `black is the default.` is not really the proper reason. It may be default now, but later it may be not. It will probably be much simple to tidy up the drawing code later if `setColor()` is left there, even if you set "default" value now.

Reply:
In the scenario you describe, it would be even better if the color is set to the default. Setting it do black would actually be worse.
But there are more things that have to be considered before we can say what is the best way. At the moment the color is not set in the majority of files when black is the wanted color after the initialization of `g (graphics)`.